service/src/main/java/com/codeforcommunity/processor/LeaderboardProcessorImpl.java

```diff
@@ -0,0 +1,96 @@
+package com.codeforcommunity.processor;
+
+import static org.jooq.generated.Tables.BLOCKS;
+import static org.jooq.generated.Tables.RESERVATIONS;
+import static org.jooq.generated.Tables.TEAMS;
+import static org.jooq.generated.tables.Users.USERS;
+import static org.jooq.impl.DSL.count;
+
+import com.codeforcommunity.api.ILeaderboardProcessor;
+import com.codeforcommunity.dto.leaderboard.GetLeaderboardRequest;
+import com.codeforcommunity.dto.leaderboard.GetLeaderboardResponse;
+import com.codeforcommunity.dto.leaderboard.LeaderboardEntry;
+import com.codeforcommunity.enums.ReservationAction;
+import java.sql.Timestamp;
+import java.util.List;
+import org.jooq.DSLContext;
+import org.jooq.Record3;
+import org.jooq.Select;
+
+public class LeaderboardProcessorImpl implements ILeaderboardProcessor {
+  private final DSLContext db;
+
+  public LeaderboardProcessorImpl(DSLContext db) {
+    this.db = db;
+  }
+
+  @Override
+  public GetLeaderboardResponse getUsersLeaderboard(
+      GetLeaderboardRequest getUsersLeaderboardRequest) {
+
+    Select<Record3<Integer, Integer, ReservationAction>> subquery =
+        db.select(BLOCKS.ID, RESERVATIONS.USER_ID, RESERVATIONS.ACTION_TYPE)
+            .distinctOn(BLOCKS.ID)
+            .from(BLOCKS)
+            .join(RESERVATIONS)
+            .onKey()
+            .where(
+                RESERVATIONS.PERFORMED_AT.greaterThan(
+                    new Timestamp(
+                        System.currentTimeMillis()
+                            - (getUsersLeaderboardRequest.getPreviousDays() * 86400000))))
```

Comment:
You should throw the 86400000 number into some static final field that explains what it represents